Sources/IonicPortals/PortalUIView.swift

```diff
@@ -144,11 +144,62 @@ public class PortalUIView: UIView {
+                """#
+
+                let handlerScript = WKUserScript(
+                    source: handlerScriptSource,
+                    injectionTime: .atDocumentStart,
+                    forMainFrameOnly: true
+                )
+
+                userContentViewController.add(PerformanceHandler(portalName: portal.name, performanceReporter: reporter), name: "vitals")
+                userContentViewController.addUserScript(handlerScript)
+            }
         }
     }
     
 }
 
+private class PerformanceHandler: NSObject, WKScriptMessageHandler {
+    let performanceReporter: WebPerformanceReporter
+    let portalName: String
+
+    init(portalName: String, performanceReporter: WebPerformanceReporter) {
+        self.performanceReporter = performanceReporter
+        self.portalName = portalName
+    }
+
+    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
+        guard let body = message.body as? [String: Any],
+              let name = body["name"] as? String,
+              let value = body["value"] as? Double
+        else { return }
+
+        switch name {
+        case "FCP":
```

Comment:
There really is no other case, but I left this open for the possibility to add more in the event WebKit/Safari get up to par with Chrome and Firefox.